presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileDecoderUtils.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.rcfile;
+
+import com.google.common.primitives.Ints;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceInput;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+// faster versions of org.apache.hadoop.io.WritableUtils methods adapted for Slice
+public final class RcFileDecoderUtils
+{
+    private RcFileDecoderUtils()
+    {
+    }
+
+    public static int decodeVIntSize(Slice slice, int offset)
+    {
+        return decodeVIntSize(slice.getByte(offset));
+    }
+
+    public static int decodeVIntSize(byte value)
+    {
+        if (value >= -112) {
+            return 1;
+        }
+        if (value < -120) {
+            return -119 - value;
+        }
+        return -111 - value;
+    }
+
+    public static boolean isNegativeVInt(Slice slice, int offset)
+    {
+        return isNegativeVInt(slice.getByte(offset));
+    }
+
+    public static boolean isNegativeVInt(byte value)
+    {
+        return value < -120 || (value >= -112 && value < 0);
+    }
+
+    public static long readVInt(SliceInput in)
+    {
+        byte firstByte = in.readByte();
+        int length = decodeVIntSize(firstByte);
+        if (length == 1) {
+            return firstByte;
+        }
+
+        long value = 0;
+        for (int i = 1; i < length; i++) {
+            value <<= 8;
+            value |= (in.readByte() & 0xFF);
+        }
+        return isNegativeVInt(firstByte) ? ~value : value;
+    }
+
+    public static long readVInt(Slice slice, int start)
+    {
+        byte firstByte = slice.getByte(start);
+        int length = decodeVIntSize(firstByte);
+        if (length == 1) {
+            return firstByte;
+        }
+
+        return readVInt(slice, start + 1, length);
+    }
+
+    public static long readVInt(Slice slice, int start, int length)
```

Comment:
I find this a dangerous public API, given it doesn't provide correct answer when `0` is provided as the `length` argument.

I would make this method private. And add a public one that handles 0. All the existing public call sites of this method will just move to the new method.